ReactComponents/CampaignView.js

```diff
@@ -60,10 +62,20 @@ var CampaignView = React.createClass({
     }
   },
   fetchData: function() {
-    var statusUrl = this.props.signupUrl + '&campaigns=' + this.props.campaign.id.toString();
+    this.setState({
+      error: false,
+      loaded: false,
+    });
+    var statusUrl = this.props.signupUrl;
+    statusUrl += '&campaigns=' + this.props.campaign.id.toString();
     fetch(statusUrl)
       .then((response) => response.json())
+      .catch((error) => this.catchError(error))
       .then((responseData) => {
+        if (!responseData) {
+          return;
```

Comment:
Without this, the code still continues even if we call the `catchError` :confused: